paddle/fluid/operators/smooth_l1_loss_v2_op.cc

```diff
@@ -0,0 +1,209 @@
+                            "inside and outside weights."));
+      auto dims = ctx->GetInputDim("InsideWeight");
+      bool check = true;
+      if ((!ctx->IsRuntime()) &&
+          (framework::product(dims) <= 0 || framework::product(x_dims) <= 0)) {
+        check = false;
+      }
+      if (check) {
+        PADDLE_ENFORCE_EQ(x_dims, dims,
+                          platform::errors::InvalidArgument(
+                              "Input(X) ans Input(InsideWeight) of "
+                              "SmoothL1LossOp should have the same size, but "
+                              "received X dim is %s, InsideWeight dim is %s",
+                              x_dims.to_str(), dims.to_str()));
+      }
+
+      dims = ctx->GetInputDim("OutsideWeight");
+      check = true;
+      if ((!ctx->IsRuntime()) &&
+          (framework::product(dims) <= 0 || framework::product(x_dims) <= 0)) {
+        check = false;
+      }
+      if (check) {
+        PADDLE_ENFORCE_EQ(x_dims, dims,
+                          platform::errors::InvalidArgument(
+                              "Input(X) ans Input(OutsideWeight) of "
+                              "SmoothL1LossOp should have the same size, but "
+                              "received X dim is %s, OutsideWeight dim is %s",
+                              x_dims.to_str(), dims.to_str()));
+      }
+    }
+
+    ctx->SetOutputDim("Diff", x_dims);
+    ctx->SetOutputDim("Out", x_dims);
+  }
+};
+
+class SmoothL1LossV2OpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X",
+             "(Tensor, default Tensor<float>) A tensor with rank at least 2. "
+             "The input value of smooth l1 loss op with shape "
+             "[batch_size, dim1, ..., dimN].");
+    AddInput("Y",
+             "(Tensor, default Tensor<float>) A tensor with rank at least 2. "
+             "The target value of smooth l1 loss op with same shape as X.");
+    AddInput("InsideWeight",
+             "(Tensor, default Tensor<float>) A tensor with rank at least 2. "
+             "This input is optional and should have same shape with X. "
+             "If provided, the result of (X - Y) will be multiplied "
+             "by this tensor element by element.")
+        .AsDispensable();
+    AddInput("OutsideWeight",
+             "(Tensor, default Tensor<float>) A tensor with rank at least 2. "
+             "This input is optional and should have same shape with X. "
+             "If provided, the out smooth l1 loss will be multiplied by this "
+             "tensor element by element.")
+        .AsDispensable();
+    AddOutput("Diff", "Intermediate variable to cache InsideWeight * (X - Y).")
+        .AsIntermediate();
+    AddOutput("Out",
+              "(Tensor, default Tensor<float>) A tensor with rank be 2. "
+              "The output smooth l1 loss has same dimension as Input(X).");
+    AddAttr<float>("sigma",
```

Comment:
default value is different with comments

Reply:
done,thx.